extensions/todoist/src/api.ts

```diff
@@ -0,0 +1,59 @@
+import { preferences, showToast, ToastStyle } from "@raycast/api";
+import axios, { AxiosError } from "axios";
+import useSWR from "swr";
+import { TaskPayload } from "./types";
+
+export const axiosInstance = axios.create({
+  baseURL: "https://api.todoist.com/rest/v1",
+  timeout: 1000,
+  headers: { Authorization: `Bearer ${preferences.token.value}` },
+});
+
+export async function createTask(body: TaskPayload): Promise<void> {
+  try {
+    await axiosInstance.post(`/tasks`, body);
+    showToast(ToastStyle.Success, "Task created");
+  } catch {
+    showToast(ToastStyle.Failure, "Could not create task");
+  }
```

Comment:
Usually we show a task while something is updated and follow this naming:

```suggestion
  const toast = showToast(ToastStyle.Animated, "Creating task");
  try {
    await axiosInstance.post(`/tasks`, body);
    toast.style = ToastStyle.Success;
    toast.title = "Created task";
  } catch (e) {
    toast.style = ToastStyle.Failure;
    toast.title = "Failed creating task"
    toast.message = e instanceof Error ? e.message : undefined; 
  }
```

Do you want to adopt it to the other actions as well?